### ISSUE TITLE
language server crashes

### ISSUE THREAD

**Issue by greg0ire** [vimeo/psalm#1562]
When I try to run the language server, here is what I get:

```
PHP Warning:  file_put_contents(/tmp/psalm/fcd9f14bfd279a5b11850b9e2668e04077b01921-s/config): failed to open stream: No such file or directory in phar:///home/gregoire/bin/psalm/src/Psalm/Internal/Provider/FileReferenceCacheProvider.php on line 254

Warning: file_put_contents(/tmp/psalm/fcd9f14bfd279a5b11850b9e2668e04077b01921-s/config): failed to open stream: No such file or directory in phar:///home/gregoire/bin/psalm/src/Psalm/Internal/Provider/FileReferenceCacheProvider.php on line 254
PHP Fatal error:  Uncaught Error: Call to undefined function _HumbugBox1b24ec6c5e06\hrtime() in phar:///home/gregoire/bin/psalm/vendor/amphp/amp/lib/Internal/functions.php:56
Stack trace:
#0 phar:///home/gregoire/bin/psalm/vendor/amphp/amp/lib/Loop/NativeDriver.php(38): _HumbugBox1b24ec6c5e06\Amp\Internal\getCurrentTime()
#1 phar:///home/gregoire/bin/psalm/vendor/amphp/amp/lib/Loop/DriverFactory.php(29): _HumbugBox1b24ec6c5e06\Amp\Loop\NativeDriver->__construct()
#2 phar:///home/gregoire/bin/psalm/vendor/amphp/amp/lib/Loop.php(413): _HumbugBox1b24ec6c5e06\Amp\Loop\DriverFactory->create()
#3 phar:///home/gregoire/bin/psalm/.box/vendor/composer/ClassLoader.php(444): include('phar:///home/gr...')
#4 phar:///home/gregoire/bin/psalm/.box/vendor/composer/ClassLoader.php(322): Composer\Autoload\includeFile('phar:///home/gr...')
#5 [internal function]: Composer\Autoload\ClassLoader->loadClass('_HumbugBox1b24e...')
#6 phar:///home/gregoire/bin/psalm/vendor/amphp/byte-stream/lib/ResourceInputStream.php(54): spl_autoload_call('_HumbugBox in phar:///home/gregoire/bin/psalm/vendor/amphp/amp/lib/Internal/functions.php on line 56

Fatal error: Uncaught Error: Call to undefined function _HumbugBox1b24ec6c5e06\hrtime() in phar:///home/gregoire/bin/psalm/vendor/amphp/amp/lib/Internal/functions.php:56
Stack trace:
#0 phar:///home/gregoire/bin/psalm/vendor/amphp/amp/lib/Loop/NativeDriver.php(38): _HumbugBox1b24ec6c5e06\Amp\Internal\getCurrentTime()
#1 phar:///home/gregoire/bin/psalm/vendor/amphp/amp/lib/Loop/DriverFactory.php(29): _HumbugBox1b24ec6c5e06\Amp\Loop\NativeDriver->__construct()
#2 phar:///home/gregoire/bin/psalm/vendor/amphp/amp/lib/Loop.php(413): _HumbugBox1b24ec6c5e06\Amp\Loop\DriverFactory->create()
#3 phar:///home/gregoire/bin/psalm/.box/vendor/composer/ClassLoader.php(444): include('phar:///home/gr...')
#4 phar:///home/gregoire/bin/psalm/.box/vendor/composer/ClassLoader.php(322): Composer\Autoload\includeFile('phar:///home/gr...')
#5 [internal function]: Composer\Autoload\ClassLoader->loadClass('_HumbugBox1b24e...')
#6 phar:///home/gregoire/bin/psalm/vendor/amphp/byte-stream/lib/ResourceInputStream.php(54): spl_autoload_call('_HumbugBox in phar:///home/gregoire/bin/psalm/vendor/amphp/amp/lib/Internal/functions.php on line 56
```

Using version 3.2.7 on a project where I just used  `psalm --init`

**Comment by muglug** [comment]
What version of PHP are you running?

**Comment by muglug** [comment]
Nevermind - I'm sure it's 7.3, and the Phar build was prefixing `hrtime` erroneously because it was built under 7.1.

**Comment by greg0ire** [comment]
Yup 7.3 :)

**Comment by greg0ire** [comment]
@muglug I know it's a bit late but I'm still experiencing this w/ version 3.4.2

```
php -v
PHP 7.3.5 (cli) (built: Apr 30 2019 21:05:09) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.3.5, Copyright (c) 1998-2018 Zend Technologies
```

**Comment by muglug** [comment]
Fixed in bfbe9a5

**Comment by greg0ire** [comment]
Yay! Thank you!
